Add release notes for v5.7.0-rc1 (#27353)
containers/podman: RELEASE_NOTES.md

- The `podman artifact rm` command now supports a new option, `--ignore`, which will suppress errors when attempting to remove an artifact that does not exist ([#27084](https://github.com/containers/podman/issues/27084)).
- The `podman artifact list` command now includes artifact creation time in its output ([#27314](https://github.com/containers/podman/issues/27314)).
- The `podman artifact list --format` option now supports a new format key, `VirtualSize`, returning the size of the artifact in integer bytes (the existing `Size` key returned human-readable information) ([#27085](https://github.com/containers/podman/issues/27085)).
- The `podman artifact inspect` command now supports a new option, `--format`, to return specific information about an artifact with user-specified formatting ([#27112](https://github.com/containers/podman/issues/27112)).

### Changes
- In preparation for a planned removal of the BoltDB database in Podman 6.0, a warning has been added for installations still using BoltDB. These warnings were added in Podman 5.6, but were not visible by default; they now are. They can be suppressed with the `SUPPRESS_BOLTDB_WARNING=true` environment variable.

> - The Windows installer has been redesigned with a simpler single MSI architecture that supports both user-scope (no admin required) and machine-scope installations. Note: Users must uninstall existing Podman installations before using the new installer, but all containers, images, machines, and other data will be preserved ([#22994](https://github.com/containers/podman/issues/22994) and [#25968](https://github.com/containers/podman/issues/25968)).

Reviewer: A small suggestion: since the old Windows installer is still here, the new one is (currently) an alternative.

```suggestion
- There is a new Windows installer, with a simpler single MSI architecture, that supports both user-scope (no admin required) and machine-scope installations. The old installer is also released for backward compatibility. Note: Users must uninstall existing Podman installations before using the new installer, but all containers, images, machines, and other data will be preserved ([#22994](https://github.com/containers/podman/issues/22994) and [#25968](https://github.com/containers/podman/issues/25968)).
```

- Podman now requires Go 1.24.
- When the `-p`/`--publish` and `--network=ns:/path` options are used together when creating a container, Podman will not warn that the `-p` option will be ignored as an existing namespace is in use (this has always been the case, but Podman now prints a warning about it) ([#26663](https://github.com/containers/podman/issues/26663)).
- The `podman stats` command now provides additional information about container resource utilization when run on FreeBSD.

> - Shell autocompletion has been enabled for the `--sysctl` option to `podman create` and `podman run` and the `--interface-name` option to `podman network create`.

Reviewer 2: ```suggestion
- Shell autocompletion has been enabled for the `--sysctl` option to `podman create` and `podman run`, and the `--interface-name` option to `podman network create`.
```

- Fixed a bug where Quadlet would fail to generate for a `.build` file when a systemd specifier was used in the `[Build]` section ([#26746](https://github.com/containers/podman/issues/26746)).
- Fixed a bug where the `podman info` command could panic when `/proc/sys/fs/binfmt_misc` was not mounted.
- Fixed a bug where the remote Podman client could lose some initial bytes of output from attach sessions (`podman run`, `podman exec`, `podman attach`) due to a race condition ([#26951](https://github.com/containers/podman/issues/26951)).
- Fixed a bug where the `podman build` command was ignoring SBOM related options ([#23915](https://github.com/containers/podman/issues/23915)).
- Fixed a bug where the `--userns=ns:/path` option to `podman create` and `podman run` was broken with runc 1.1.11 and higher ([#27148](https://github.com/containers/podman/issues/27148)).

### API
- Added a new API endpoint to list quadlets (`GET /libpod/quadlets/json`).
- The Compat Inspect endpoint for Images no longer includes the `ContainerConfig` field. To access image configuration, use the `Config` field instead. This matches changes made by Docker in the v1.45 API.

> - Fixed a bug where the Stats and Commit endpoints for Containers (compat & libpod), Push, Commit, Push, and Pull endpoints for Images (compat & libpod), and Push endpoint for Manifests (libpod) were not returning a `Content-Type` header.

Reviewer 2: ```suggestion
- Fixed a bug where the Stats and Commit endpoints for Containers (compat & libpod), Push, Commit, and the Pull endpoints for Images (compat & libpod), and the Push endpoint for Manifests (libpod) were not returning a `Content-Type` header.
```

### Misc
- Error messages returned when an incomplete `--device` option (e.g. `--device /dev/fuse::`) is passed to `podman create` or `podman run` have been improved.
- Updated Buildah to v1.42.0
- Updated the containers/image library to v5.38.0
- Updated the containers/storage library to v1.61.0
- Updated the containers/common library to v0.66.0

> - The containers/image, containers/storage, and containers/common libraries are now sourced from the [containers/container-libs](https://github.com/containers/container-libs/) monorepo.

Reviewer 2: Maybe, or maybe not, your call.:
```suggestion
- The containers/image, containers/storage, and containers/common libraries are now sourced from the [containers/container-libs](https://github.com/containers/container-libs/) monorepo.  As a result, the vendored entries have changed from github.com/containers/common to go.podman.io/common, for example.
```

## 5.6.2
### Bugfixes

> - Fixed a bug where stopping the `podman machine start` command with SIGPIPE could result in machine state being stuck as "Starting" ([#26949](https://github.com/containers/podman/issues/26949)).

Reviewer 2: ```suggestion
- Fixed a bug where stopping the `podman machine start` command with SIGPIPE could result in the machine state being stuck as "Starting" ([#26949](https://github.com/containers/podman/issues/26949)).
```

- Fixed a bug where `podman build` would fail with a permissions error when building Containerfiles using a non-root user and cache mounts ([#27044](https://github.com/containers/podman/issues/27044)).

### Misc
- Updated Buildah to v1.41.5

## 5.6.1
### Security
- This release addresses CVE-2025-9566, where Kubernetes YAML run by `podman play kube` containing `ConfigMap` and `Secret` volumes can use crafted symlinks to overwrite content on the host.

### Bugfixes
- Fixed a bug where network creation and removal events were displayed incorrectly when the `journald` events driver was in use.
- Fixed a bug where the `--security-opt seccomp=unconfined` option was broken on Windows ([#26855](https://github.com/containers/podman/issues/26855)).

> - Fixed a bug where containers created with a name longer than 64 characters, no explicit hostname, the the `container_name_as_hostname` option in `containers.conf` set to `true` would fail to start.

Reviewer 2: ```suggestion
- Fixed a bug where containers created with a name longer than 64 characters, no explicit hostname, the  `container_name_as_hostname` option in `containers.conf` set to `true` would fail to start.
```
